website/src/blogs/how-to-deploy-jenkins-on-kubernetes-openebs.md

```diff
@@ -0,0 +1,222 @@
+---
+title: How to deploy Jenkins on Kubernetes + OpenEBS
+author: Murat Karslioglu
+author_info: VP @OpenEBS & @MayaData_Inc. Lives to innovate! Opinions my own!
+date: 16-11-2017
+tags: Jenkins, Kubernetes, Open Source, Solutions, Openebs
+excerpt: Modern development requires Continuous Integration / Continuous Delivery (CI/CD) and it means building and validating your software on every commit to make sure your development & test environments are always up-to-date. 
+---
+
+Modern development requires [Continuous Integration](https://aws.amazon.com/devops/continuous-integration/) / [Continuous Delivery](https://aws.amazon.com/devops/continuous-delivery/) (**CI/CD**) and it means building and validating your software on every commit to make sure your development & test environments are always up-to-date. This level of automation is a combination of cultural philosophies (aka **DevOps**) and practices. CI/CD increases an organization’s ability to deliver applications and services at high velocity. Jenkins serves as the **workflow engine** to manage this **CI/CD pipeline** from source to delivery.
+
+Deploying Jenkins on Kubernetes provides the following benefits:
+
+- Isolates different jobs from one another
+- Quickly clean a job’s workspace
+- Dynamically deploy or schedule jobs with Kubernetes pods
+- Allows increased resource utilization and efficiency
+- Dynamically scale up Jenkins slaves on demand
+
+Especially, running dynamic slaves in a Kubernetes/Docker environment and automating the scaling of Jenkins slaves running in Kubernetes on top of OpenEBS can **minimize the deployment time and cost**. With OpenEBS, you can build extremely scalable test cycles. You will be able to create instant snapshots (thanks to the [**CoW**](https://en.wikipedia.org/wiki/Copy-on-write)) from the master and **deploy new slaves faster and dynamically on demand**. This process will eliminate the need to perform container-to-container copies.
+
+In Jenkins, slaves are optional. OpenEBS can also help when you have a smaller environment and running a **monolithic master**. In that model, state on the master would be lost when you shut down the Jenkins master service. When using monolithic master on OpenEBS, your volume is persistent and replicated over to n nodes (defined in your OpenEBS storage class). In that case, the master can exit, even if your node fails it can start on other nodes, migrate from private to public cloud, vice-versa when needed and your data will follow you.
```

Comment:
```suggestion
In Jenkins, slaves are optional. OpenEBS can also help when you have a smaller environment and running a **monolithic master**. In that model, state on the master would be lost when you shut down the Jenkins master service. When using monolithic master on OpenEBS, your volume is persistent and replicated over to n nodes *(defined in your OpenEBS storage class)*. In that case, the master can exit, even if your node fails it can start on other nodes, migrate from private to public cloud, vice-versa when needed and your data will follow you.
```